tests/test_unit.py

```diff
@@ -99,8 +100,163 @@ def request(self, *args, **kwargs):
 
 
 def test_tensorboard():
+    """
+    Test that tensorboard.SummaryWriter object is created only when folder path is passed
+    """
     model = Spec2Pep(tb_summarywriter="test_path")
     assert model.tb_summarywriter is not None
 
     model = Spec2Pep()
     assert model.tb_summarywriter is None
+
+
+def test_beam_search_decode():
+    """
+    Test that beam search decoding and its sub-functions as intended during inference
+    """
+    model = Spec2Pep(n_beams=4)
```

Comment:
**issue:** Can we have some small tests with different numbers of beams (1 should be there for sure).

Reply:
Added `n_beams=1` but couldn't think of other meaningful **small** tests to cover other beam sizes.